structure_tests/ext_run.sh

```diff
@@ -74,8 +74,20 @@ while test $# -gt 0; do
 				# the docker volume mount, we copy all of these configs into
 				# a /tmp directory and mount this single directory into the
 				# test image. this directory is cleaned up after testing.
-				cp "$1" "$CONFIG_DIR"/cfg_$CONFIG_COUNTER.json
-				CMD_STRING=$CMD_STRING" --config /cfg/cfg_$CONFIG_COUNTER.json"
+				case "$1" in
+					*json)
+						EXTENSION=".json"
+						;;
+					*yaml)
+						EXTENSION=".yaml"
+						;;
+					*)
+						echo "please provide valid JSON or YAML file: $1"
+						exit 1
+						;;
+				esac
+				cp "$1" "$CONFIG_DIR"/cfg_$CONFIG_COUNTER$EXTENSION
```

Comment:
Could we simplify this by just copying the entire file name? Then you wouldn't need to do the counters either.

I think this is possible now that we just cleanup the entire tmpdir.